packages/cache/src/internal/tar.ts

```diff
@@ -26,6 +26,8 @@ async function getTarPath(
     case 'darwin': {
       const gnuTar = await io.which('gtar', false)
       if (gnuTar) {
+        // fix permission denied errors when extracting BSD tar archive with GNU tar - https://github.com/actions/cache/issues/527
+        args.push('--delay-directory-restore')
```

Comment:
Changes look good, however we have a test that verifies what args these are called with you will need to modify.

Reply:
This does unfortunately append this arg to all invocations of `tar`, including list and create, but appears to be silently ignored in those cases.

Reply:
Tests are fixed